File: objectModel/Python/cdm/persistence/cdmfolder/projections/operation_array_expansion_persistence.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import TYPE_CHECKING

from cdm.enums import CdmObjectType
from cdm.persistence.cdmfolder.types import OperationArrayExpansion

from .operation_base_persistence import OperationBasePersistence

if TYPE_CHECKING:
    from cdm.objectmodel import CdmCorpusContext, CdmOperationArrayExpansion
    from cdm.utilities import ResolveOptions, CopyOptions


class OperationArrayExpansionPersistence:
    """Operation ArrayExpansion persistence"""

    @staticmethod
    def from_data(ctx: 'CdmCorpusContext', data: 'OperationArrayExpansion') -> 'CdmOperationArrayExpansion':
        if not data:
            return None
        
        array_expansion_op = OperationBasePersistence.from_data(ctx,
            CdmObjectType.OPERATION_ARRAY_EXPANSION_DEF, data)  # type: CdmOperationArrayExpansion
        array_expansion_op.start_ordinal = data.startOrdinal
        array_expansion_op.end_ordinal = data.endOrdinal

        return array_expansion_op

    @staticmethod
    def to_data(instance: 'CdmOperationArrayExpansion', res_opt: 'ResolveOptions', options: 'CopyOptions') -> 'OperationArrayExpansion':
        if not instance:
            return None

        obj = OperationBasePersistence.to_data(instance, res_opt, options)  # type: OperationArrayExpansion
        obj.startOrdinal = instance.start_ordinal
        obj.endOrdinal = instance.end_ordinal

        return obj

File: objectModel/TypeScript/Persistence/CdmFolder/Projections/OperationArrayExpansionPersistence.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusContext,
    cdmObjectType,
    CdmOperationArrayExpansion,
    copyOptions,
    resolveOptions,
} from '../../../internal';
import { OperationArrayExpansion } from '../types';
import { OperationBasePersistence } from './OperationBasePersistence';

/**
 * Operation ArrayExpansion persistence
 */
export class OperationArrayExpansionPersistence {
    public static fromData(ctx: CdmCorpusContext, object: OperationArrayExpansion): CdmOperationArrayExpansion {
        if (!object) {
            return undefined;
        }

        const arrayExpansionOp: CdmOperationArrayExpansion = OperationBasePersistence.fromData(ctx, cdmObjectType.operationArrayExpansionDef, object);
        arrayExpansionOp.startOrdinal = object.startOrdinal;
        arrayExpansionOp.endOrdinal = object.endOrdinal;

        return arrayExpansionOp;
    }

    public static toData(instance: CdmOperationArrayExpansion, resOpt: resolveOptions, options: copyOptions): OperationArrayExpansion {
        if (!instance) {
            return undefined;
        }

        const data: OperationArrayExpansion = OperationBasePersistence.toData(instance, resOpt, options);
        data.startOrdinal = instance.startOrdinal;
        data.endOrdinal = instance.endOrdinal;

        return data;
    }
}
